community/history.md

```diff
@@ -0,0 +1,123 @@
+
+Here around 2017, Continuum renamed itself to Anaconda, so let's switch those names from here out.
+
+As more and more conflicts with `free` channel packages occurred, conda-forge gradually added more and more of their own core dependency packages to avoid those breakages. At the same time, Anaconda was working on two contracts that would prove revolutionary. Samsung wanted to use conda packages to manage their internal toolchains, and Ray suggested that this was complementary to our own internal needs to update our toolchain. Samsung's contract supported development to conda-build that greatly expanded its ability to support explicit variants of recipes. Intel was working on developing their own Python distribution at the time, which they based on Anaconda and added their accelerated math libraries and patches to. Part of the Intel contract was that Anaconda would move all of their internal recipes into public-facing GitHub repositories. Rather than putting another set of repositories (another set of changes to merge) in between internal and external sources, such as conda-forge, Michael and Ray pushed for a design where conda-forge would be the reference source of recipes. Anaconda would only carry local changes if they were not able to be incorporated into the conda-forge recipe for social, licensing, or technical reasons. The combination of these conda-forge based recipes and the new toolchain are what made up the `main` channel, which was also part of `defaults`. The `main` channel represented a major step forward in keeping conda-forge and Anaconda aligned, which equates to smooth operation and happy users. The joined recipe base and toolchain has sometimes been contentious, with conda-forge wanting to move faster than Anaconda or vice-versa. The end result has been a compromise between cutting-edge development and slower enterprise-focused development.
```

Comment:
AFAIR, the compiler stack between defaults and conda-forge wasn't actually shared until GCC 7 I think? I realize though that that was comparatively smaller work&impact than the decision to start building the compilers ourselves (kudos in retrospect 👏)

(also, could we break these really long lines a bit?)

Reply:
GCC 7 is what the first compiler-in-a-package was. Ray knew that we had to keep on the cutting edge if we were going to ship our own libstdc++. If load order led to an older, conda-based libstdc++, then some later library being loaded may not find the (system) symbols that it needs. In practice, the compiler stack hasn't been kept on the cutting edge consistently. It hasn't been a huge issue, since conda packages shouldn't be reaching out to the system libraries, but the risk is there.

Reply:
> In practice, the compiler stack hasn't been kept on the cutting edge consistently.

It has in recent years 😊 

(we intentionally keep the _default_ compiler version in the pinning roughly 9-15 months behind their respective newest releases, but new compiler versions are published on the order of weeks after release for GCC, and less than that for clang; in particular, since the C++ stdlibs are only constrained to >= compiler_version, this means our libgcc/libstdcxx/libcxx etc. stay on the cutting edge consistently.)

Reply:
> GCC 7 is what the first compiler-in-a-package was.

Hm, I can find [packages for 5.4](https://anaconda.org/main/gcc_impl_linux-64/files?version=5.4.0). Is is then true that GCC 7 was the first?

Reply:
I honestly have no idea what that GCC5 package is from. Too many beers between then and now.

What I can say is that the timestamps on repo.anaconda.com indicate that the GCC 7 packages were available a few months before the single GCC 5 one.

<img width="886" alt="Screenshot 2025-04-14 at 10 17 08 PM" src="https://github.com/user-attachments/assets/ef72d033-debd-4be5-9219-035c239e4c2b" />

Ray was absolutely insistent that GCC 7 (newest at the time we started the work) be our toolchain.